extensions/sql/contract-definition/loader/src/test/java/org/eclipse/dataspaceconnector/sql/contractdefinition/loader/SqlContractDefinitionLoaderTest.java

```diff
@@ -0,0 +1,138 @@
+                SqlContractDefinitionLoaderTest.this.dataSourceRegistry = context.getService(DataSourceRegistry.class);
+                SqlContractDefinitionLoaderTest.this.contractDefinitionLoader = context.getService(ContractDefinitionLoader.class);
+
+                try (var inputStream = this.getClass().getClassLoader().getResourceAsStream("schema.sql")) {
+                    var schema = new String(Objects.requireNonNull(inputStream).readAllBytes(), StandardCharsets.UTF_8);
+                    try (var connection = dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection()) {
+                        transactionContext.execute(() -> SqlQueryExecutor.executeQuery(connection, schema));
+                    }
+                } catch (IOException | SQLException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+    }
+
+    @AfterEach
+    void tearDown() {
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection()) {
+                executeQuery(connection, String.format("DELETE FROM %s", SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE));
+            } catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    @DisplayName("Context Loads, tables exist")
+    void contextLoads() throws SQLException {
+        var query = String.format("SELECT 1 FROM %s", SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE);
+        executeQuery(dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection(), query);
+    }
+
+    @Test
+    @DisplayName("Accepts and persists ContractDefinition")
+    void testAcceptEntry() throws SQLException {
+        var accessPolicy = Policy.Builder.newInstance()
+                .id("access")
+                .build();
+        var contractPolicy = Policy.Builder.newInstance()
+                .id("contract")
+                .build();
+        var expression = AssetSelectorExpression.Builder.newInstance()
+                .build();
+        var definition = ContractDefinition.Builder.newInstance()
+                .id("id")
+                .accessPolicy(accessPolicy)
+                .contractPolicy(contractPolicy)
+                .selectorExpression(expression)
+                .build();
+
+        contractDefinitionLoader.accept(definition);
+
+        var query = String.format("SELECT count(*) FROM %s WHERE %s=?",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID);
+
+        var definitionCount = executeQuery(
+                dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection(),
+                (rs) -> rs.getLong(1),
+                query, "id").iterator().next();
+
+        assertThat(definitionCount).isEqualTo(1);
+    }
```

Comment:
a few test cases are not covered: `accept_alreadyExists`, `accept_differentId_sameParameters`, `accept_exceptionDuringSql", etc. Please add.